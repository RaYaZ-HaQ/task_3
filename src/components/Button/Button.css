.button {
  border-radius: 9999px; /* Rounded ends */
  background-color: #c3c3e5;
  padding: 0.25rem 0.75rem;
  font-size: 1rem;
  font-family: Arial, Helvetica, sans-serif;
  letter-spacing: 0.075rem;
  width: fit-content;
}

.button:hover {
  cursor: pointer;
  background-color: #c5c5f5;
}

.spin {
  animation: rotate-cw infinite 5s linear;
}

@keyframes rotate-cw {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
