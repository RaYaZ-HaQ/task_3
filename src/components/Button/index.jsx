import "./Button.css";
import CheckIcon from "../../assets/icons/CheckIcon";
import LoaderIcon from "../../assets/icons/LoaderIcon";

/* eslint-disable react/prop-types */
export default function Button({
  children,
  onClick,
  loading = false,
  success = false,
}) {
  return (
    <button
      onClick={onClick}
      className="button flex-container items-center border-black"
    >
      {!loading && success && <CheckIcon strokeColor="green" />}
      {loading && <LoaderIcon className="spin" />}
      <span style={{ marginLeft: loading || success ? "0.5rem" : "" }}>
        {children}
      </span>
    </button>
  );
}
