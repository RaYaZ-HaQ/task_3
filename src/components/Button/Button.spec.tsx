import React from "react";
import { render, screen } from "@testing-library/react";
import Button from ".";

describe("Button tests", () => {
  it("should match snapshot", () => {
    const { container } = render(<Button>Button text</Button>);

    expect(container).toMatchSnapshot();
  });

  it("should show loading icon when loading", async () => {
    render(<Button loading>Button text</Button>);

    const loader = await screen.findByTestId("loading-icon");
    expect(loader).toBeTruthy();
  });

  it("should show check mark icon when success", async () => {
    render(<Button success>Button text</Button>);

    const successIcon = screen.getByTestId("check-icon");
    expect(successIcon).toBeTruthy();
    expect(successIcon.getAttribute("stroke")).toEqual("green");
  });
});
