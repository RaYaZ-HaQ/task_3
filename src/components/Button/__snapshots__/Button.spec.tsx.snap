// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match snapshot 1`] = `
<div>
  <button
    class="button flex-container items-center border-black"
  >
    <span>
      Button text
    </span>
  </button>
</div>
`;
